### PR TITLE
Fix "Using the material on more than one mesh breaks it"

### PR DESCRIPTION
**Pull request by johnmarinelli**
it appears that `DEFAULT_STATE` was getting modified when it shouldn't have been, the solution is to clone it as necessary

### CODE REVIEW

File: src/component-material.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ export const ComponentMaterial = React.forwardRef(function ComponentMaterial(
         }
 
         return acc
-      }, DEFAULT_STATE),
+      }, Object.assign({}, DEFAULT_STATE)),
```
Comment (gsimone) [inline src/component-material.tsx]: Should we use the more modern syntax with `{ ...DEFAULT_STATE }` ?

Comment (johnmarinelli) [reply src/component-material.tsx]: sure, i'll update it

File: src/component-material.tsx [hunk 2]
```diff
@@ -70,40 +70,43 @@ export const ComponentMaterial = React.forwardRef(function ComponentMaterial(
 
   const shaders = useMemo<ExtensionShadersObject>(
     () =>
-      React.Children.toArray(children).reduce((acc: any, child: any) => {
-        const shader = child?.props?.children
+      React.Children.toArray(children).reduce(
```
Comment (johnmarinelli) [inline src/component-material.tsx:73]: these are all Prettier's changes

File: src/component-material.tsx [hunk 3]
```diff
@@ -70,40 +70,43 @@ export const ComponentMaterial = React.forwardRef(function ComponentMaterial(
...
               `)
+            }
           }
-        }
 
-        return acc
-      }, DEFAULT_STATE),
+          return acc
+        },
+        { ...DEFAULT_STATE }
```
Comment (johnmarinelli) [inline src/component-material.tsx:108]: here is the line that actually changed

**Comment by emmelleppi** [comment]
Hi @johnmarinelli ! First of all thank you for this PR :)
Sorry if I answer you just now, but I only picked up again this project this weekend.

I'm testing your PR but it keeps giving me the same uniform redefinition error.

Does it work without problems for you?

**Comment by johnmarinelli** [comment]
@emmelleppi it did when i had tried it, but let me try again and get back to you

**Comment by johnmarinelli** [comment]
closing in favor of this https://github.com/pmndrs/component-material/pull/14
